### PR TITLE
Fix: update toolchain clippy, fix clippy. Fix vulnarabilities

### PR DESCRIPTION
**Pull request by mrLSD**
## Description

➡️ Updated rust toolchain to `1.80.0`
➡️ Added `clippy::pedantic`
➡️ Fixed new clippy issues 
⚠️ Found and fixed vulnerabilities related to `signed->unsigned` casting. It affects `gas_used` calculations. In some cases, it was possible cases, when casting `i64->u64` could produce an unexpected number, and `gas_used` calculation may produce wrong result.
➡️ Fixed casting logic for `u64->usize` as it can produce truncated numbers fro 32-bit systems (like `wasm32`)
➡️ Improved numbers casting logic
➡️ Added `clippy::as_conversions`

### CODE REVIEW

**Review by aleksuss — APPROVED** [review]
LGTM, but also it would be good to add a description about why an error occurs in the #Errors section in the doc-comments. But it's probably out of the scope of this PR.

File: runtime/src/lib.rs [hunk 1]
```diff
@@ -445,7 +465,8 @@ impl Config {
 
 		// See https://eips.ethereum.org/EIPS/eip-3529
 		let refund_sstore_clears = if decrease_clears_refund {
-			(gas_sstore_reset + gas_access_list_storage_key) as i64
+			// Avoid signed casting to unsigned as it'll overflow
```
Comment (birchmd) [inline runtime/src/lib.rs:468]:
```suggestion
			// Avoid unsigned casting to signed as it may overflow
```

File: src/executor/stack/executor.rs [hunk 2]
```diff
@@ -231,42 +253,59 @@ pub trait StackState<'config>: Backend {
...
 	/// Get tstorage value of address at index.
 	/// EIP-1153: Transient storage
+	///
+	/// # Errors
+	/// Return `ExitError`
 	fn tload(&mut self, address: H160, index: H256) -> Result<U256, ExitError>;
 }
 
-/// Stack-based executor.
+/// Stack-based executor.\
```
Comment (birchmd) [inline src/executor/stack/executor.rs:308]:
```suggestion
/// Stack-based executor.
```

**Review by birchmd — APPROVED** [review]
Great job!

**Pull request merged**